BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
```

Comment:
- **Constants** should use PascalCase by convention, but in this case, the question here is that the **URL is hardcoded**, because everytime we'll need to make any changes, we would have to deploy the application. 
- To improve maintainability, I strongly suggest to use the **Options Pattern**, that will get the values from the **appsettings.json**. You could have a class called **CnbOptions** with the same structure that you'll have in the .json file.

For more info: https://learn.microsoft.com/en-us/dotnet/core/extensions/options

Reply:
- According to the requirements, this service will initially target only hotels that use CZK. For this reason, we could use a **feature flag** like `AllowAllSourceCurrencies`  inside of the **ExchangeRateService**. From the appsetting.json, we can activate and deactivate it to allow all source currencies.
* In this case, we will probably need to retrieve the currency (or the currencies) used by the hotel making the request to determine whether to process it or not. For example: if `AllowAllSourceCurrencies` is set to **false**, we could, for example, return an error when hotels who don't use CZK try to use the service. For more information: https://learn.microsoft.com/en-us/azure/azure-app-configuration/use-feature-flags-dotnet-core